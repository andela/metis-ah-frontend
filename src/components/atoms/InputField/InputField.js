import React from 'react';

const input = ({ name, label, ...props }) => (
  <label htmlFor={name}>
    {label}
    <input name={name} {...props} />
  </label>
);
export default input;
