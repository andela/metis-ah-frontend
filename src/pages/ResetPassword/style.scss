
#reset-password-container {
  box-shadow: 0 1px 2px 2px rgba(0,0,0,0.09);
  margin: 120px auto;
  width: 500px;
  border-radius: 4px;

  h4 {
    border-bottom: 1px solid rgb(240, 238, 238);
    color: #4dd6a5;
    display: block;
    width: 100%;
    padding: 10px;
    text-align: center;
  }

  form {
    padding: 40px 20px;
    width: 100%;
    border: 0;
  }

  input#email-input {
    border: 1px solid #4dd6a5;;
    display: block;
    width: 100%;
    height: 35px;
    font-size: 18px;
    padding: 5px;
    margin-top: 10px;
    border-radius: 4px;
  }
  input[type="submit"] {
    border: 0;
    background: #4dd6a5;
    color: white;
    font-weight: 600;
    margin-top: 8px;
    padding: 10px 30px;
    border-radius: 3px;
    cursor: pointer;
  }
  input[type="submit"]:hover {
    background: #11b179;
  }

  #error-message {
    text-align: center;
    display: block;
    width: 100%;
    background: rgb(241, 59, 59);
    color: white;
  }
  #success-message {
    text-align: center;
    display: block;
    width: 100%;
    background: #11b179;
    color: white;
  }
}
