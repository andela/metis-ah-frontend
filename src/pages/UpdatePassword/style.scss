#reset-password-final-body {
  padding-bottom: 125px;
  background: rgb(246, 246, 246);

  #reset-password-container {
    box-shadow: 0 1px 2px 2px rgba(0,0,0,0.09);
    margin: 120px auto;
    width: 500px;
    border-radius: 4px;
    background: white;

    h4 {
      border-bottom: 1px solid rgb(240, 238, 238);
      color: #4dd6a5;
      display: block;
      width: 100%;
      padding: 10px;
      text-align: center;
    }

    form {
      padding: 40px 20px;
      width: 100%;
      border: 0;
    }

    input.password-input {
      border: 1px solid #4dd6a5;
      display: block;
      width: 100%;
      height: 35px;
      font-size: 18px;
      padding: 5px;
      margin-bottom: 10px;
      border-radius: 4px;
    }
    input[type="submit"] {
      border: 0;
      background: #4dd6a5;
      color: white;
      font-weight: 600;
      margin-top: 8px;
      padding: 10px 30px;
      border-radius: 20px;
      cursor: pointer;
      font-size: 16px;
    }
    input[type="submit"]:hover {
      background: #11b179;
    }
    label {
      font-size: 14px;
    }

    #spinner {
      position: absolute;
      bottom: 50px;
      left: 140px;
      color: #11b179;
    }
  }

  @media only screen and (max-width: 510px) {
    #reset-password-container {
      width: 300px;

      form {
        padding: 40px 8px;
      }
      input.password-input {
        height: 30px;
        font-size: 15px;
      }
    }
  }
}
