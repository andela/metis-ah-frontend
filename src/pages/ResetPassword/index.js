import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { resetPassword } from 'Actions/resetPasswordActions';
import FooterBrand from 'Components/compounds/FooterBrand';
import validateEmail from 'Utils/validateEmail';
import BasicNav from 'Components/compounds/BasicNav';
import InputField from 'Components/atoms/InputField/InputField';

import Alert from 'Utils/helpers/alert';
import logo from 'Images/logo.png';
import './style.scss';

export class ResetPassword extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      email: ''
    };
  }

  handleChange = (e) => {
    this.setState({
      email: e.target.value
    });
  }

  handleSubmit = (e) => {
    e.preventDefault();
    const { email } = this.state;
    const { resetPassword } = this.props;

    const { isValidData, error } = validateEmail(email);
    if (isValidData) {
      return resetPassword(email);
    }
    return Alert.error(error);
  }

  render() {
    const { email } = this.state;
    const { isLoading } = this.props;
    return (
      <Fragment>
        <div id="reset-password-body">

          <BasicNav
            brandName="Author&apos;s Haven"
            logo={logo}
          />

          <div className="container" id="reset-password-container">
            <h4 id="psw-title">RESET PASSWORD</h4>

            <form onSubmit={this.handleSubmit}>
              <InputField onChange={this.handleChange} id="email-input" type="email" value={email} name="email" label="Enter Your Email" />
              <InputField type="submit" value="SEND" style={isLoading ? { opacity: 0.3 } : null} disabled={isLoading} />
              <i id="spinner" className={isLoading ? 'fas fa-spinner fa-spin' : ''} />
            </form>
          </div>
        </div>

        <FooterBrand />
      </Fragment>

    );
  }
}


ResetPassword.propTypes = {
  resetPassword: PropTypes.func.isRequired,
  isLoading: PropTypes.bool.isRequired
};

const mapStateToProps = state => ({
  isLoading: state.passwordReset.isLoading
});

export default connect(
  mapStateToProps,
  { resetPassword }
)(ResetPassword);
