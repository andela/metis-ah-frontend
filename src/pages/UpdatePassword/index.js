import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import FooterBrand from 'Components/compounds/FooterBrand';
import BasicNav from 'Components/compounds/BasicNav';
import InputField from 'Components/atoms/InputField/InputField';
import { resetVerifiedUserPassword } from 'Actions/resetPasswordActions';
import validatePassword from 'Utils/validatePassword';
import Alert from 'Utils/helpers/alert';
import logo from 'Images/logo.png';
import './style.scss';

export class UpdatePassword extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      userInfo: {
        password: '',
        confirmPassword: ''
      }
    };
  }

  handleChange = (e) => {
    const { userInfo } = this.state;
    const newUserInfo = { ...userInfo };
    newUserInfo[`${e.target.name}`] = e.target.value;
    this.setState({
      userInfo: newUserInfo
    });
  }

  handleSubmit = (e) => {
    e.preventDefault();
    const { userInfo } = this.state;
    const { resetVerifiedUserPassword } = this.props;
    const { isValidData, error } = validatePassword(userInfo);

    if (isValidData) {
      const { match, history } = this.props;
      return resetVerifiedUserPassword(userInfo.password, match.params.token, history);
    }
    return Alert.error(error);
  }

  render() {
    const { userInfo } = this.state;
    const { isLoading } = this.props;
    return (
      <Fragment>
        <div id="reset-password-final-body">

          <BasicNav
            brandName="Author&apos;s Haven"
            logo={logo}
          />

          <div className="container" id="reset-password-container">
            <h4 id="psw-title">RESET PASSWORD</h4>
            <form onSubmit={this.handleSubmit}>
              <InputField label="New Password" onChange={this.handleChange} name="password" className="password-input" type="password" value={userInfo.password} />
              <InputField label="Confirm New Password" onChange={this.handleChange} name="confirmPassword" className="password-input" type="password" value={userInfo.confirmPassword} />
              <InputField type="submit" value="SAVE" style={isLoading ? { opacity: 0.3 } : null} disabled={isLoading} />
              <i id="spinner" className={isLoading ? 'fas fa-spinner fa-spin' : ''} />
            </form>
          </div>
        </div>

        <FooterBrand />
      </Fragment>
    );
  }
}

UpdatePassword.propTypes = {
  resetVerifiedUserPassword: PropTypes.func.isRequired,
  match: PropTypes.objectOf(PropTypes.any).isRequired,
  history: PropTypes.objectOf(PropTypes.any).isRequired,
  isLoading: PropTypes.bool.isRequired
};

const mapStateToProps = state => ({
  isLoading: state.passwordReset.isLoading
});

export default connect(
  mapStateToProps,
  { resetVerifiedUserPassword }
)(UpdatePassword);
