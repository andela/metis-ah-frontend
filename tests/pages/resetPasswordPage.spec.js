/* global jest */

import React from 'react';
import { shallow } from 'enzyme';

import { ResetPassword } from 'Pages/ResetPassword';
import { UpdatePassword } from 'Pages/UpdatePassword';
import InputField from '../../src/components/atoms/InputField/InputField';


let wrapper;
const mockFn = jest.fn();
describe('ResetPassword page', () => {
  beforeEach(() => {
    const props = {
      isLoading: false,
      resetPassword: mockFn
    };
    wrapper = shallow(<ResetPassword {...props} />);
  });

  test('should contain the following elements', () => {
    expect(wrapper.exists()).toBe(true);
    expect(wrapper.find(InputField).exists()).toBe(true);
    expect(wrapper.find('form').exists()).toBe(true);
    expect(wrapper.find('h4').exists()).toBe(true);
    expect(wrapper.find('h4').text()).toEqual('RESET PASSWORD');

    expect(wrapper.find('#spinner').exists()).toBe(true);
  });

  test('should call handleSubmit method on form submit', () => {
    const event = {
      preventDefault: mockFn
    };
    wrapper.instance().handleSubmit = mockFn;
    wrapper.update();

    wrapper.find('form').simulate('submit', event);
    expect(wrapper.instance().handleSubmit).toHaveBeenCalled();
  });

  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should set email on input change', () => {
    const event = {
      target: {
        name: 'email',
        value: 'johngorithm@gmail.com'
      }
    };
    const email = wrapper.find('#email-input');
    email.simulate('change', event);

    expect(wrapper.instance().state.email).toBe('johngorithm@gmail.com');
  });

  test('should submit form', () => {
    const event = {
      preventDefault: mockFn
    };
    wrapper.setState({
      email: 'johngorithm@gmail.com'
    });

    const form = wrapper.find('form');
    form.simulate('submit', event);

    expect(wrapper.instance().props.resetPassword).toHaveBeenCalledWith('johngorithm@gmail.com');
    expect(wrapper.instance().state.email).toBe('johngorithm@gmail.com');
  });

  test('should should spinner when loading', () => {
    wrapper.setProps({ isLoading: true });
    expect(wrapper.instance().props.isLoading).toBe(true);
  });
});

describe('ResetPasswords page', () => {
  beforeEach(() => {
    const props = {
      isLoading: false,
      resetVerifiedUserPassword: mockFn,
      match: {
        params: {
          token: 'k4tkwertlk554kjlkh4235lkjl6ghjgf56fhf352'
        }
      },
      history: {}
    };
    wrapper = shallow(<UpdatePassword {...props} />);
  });

  test('should contain the following elements', () => {
    expect(wrapper.exists()).toBe(true);
    expect(wrapper.find(InputField).exists()).toBe(true);
    expect(wrapper.find(InputField)).toHaveLength(3);
    expect(wrapper.find('form').exists()).toBe(true);
    expect(wrapper.find('h4').exists()).toBe(true);
    expect(wrapper.find('h4').text()).toEqual('RESET PASSWORD');

    expect(wrapper.find('#spinner').exists()).toBe(true);
  });

  test('should call handleSubmit method on form submit', () => {
    const event = {
      preventDefault: mockFn
    };
    wrapper.instance().handleSubmit = mockFn;
    wrapper.update();

    wrapper.find('form').simulate('submit', event);
    expect(wrapper.instance().handleSubmit).toHaveBeenCalled();
  });

  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should set password on input change', () => {
    const event = {
      target: {
        name: 'password',
        value: 'PassWord'
      }
    };
    const passwordInput = wrapper.find('[name="password"]');
    passwordInput.simulate('change', event);

    expect(wrapper.instance().state.userInfo.password).toEqual('PassWord');
  });

  test('should submit form', () => {
    const event = {
      preventDefault: mockFn
    };

    const userInfo = {
      password: 'PassWord',
      confirmPassword: 'PassWord'
    };
    wrapper.setState({
      userInfo
    });

    const form = wrapper.find('form');
    form.simulate('submit', event);

    expect(wrapper.instance().props.resetVerifiedUserPassword).toHaveBeenCalledWith('PassWord', 'k4tkwertlk554kjlkh4235lkjl6ghjgf56fhf352', {});
    expect(wrapper.instance().state.userInfo.password).toBe('PassWord');
  });

  test('should should spinner when loading', () => {
    wrapper.setProps({ isLoading: true });
    expect(wrapper.instance().props.isLoading).toBe(true);
  });
});
