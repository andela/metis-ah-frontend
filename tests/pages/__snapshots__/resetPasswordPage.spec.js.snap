// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPassword page should match snapshot 1`] = `
<Fragment>
  <div
    id="reset-password-body"
  >
    <BasicNav
      brandName="Author's Haven"
      logo="test-file-stub"
    />
    <div
      className="container"
      id="reset-password-container"
    >
      <h4
        id="psw-title"
      >
        RESET PASSWORD
      </h4>
      <form
        onSubmit={[Function]}
      >
        <InputField
          id="email-input"
          label="Enter Your Email"
          name="email"
          onChange={[Function]}
          type="email"
          value=""
        />
        <InputField
          disabled={false}
          label=""
          name=""
          style={null}
          type="submit"
          value="SEND"
        />
        <i
          className=""
          id="spinner"
        />
      </form>
    </div>
  </div>
  <Brand />
</Fragment>
`;

exports[`ResetPasswords page should match snapshot 1`] = `
<Fragment>
  <div
    id="reset-password-final-body"
  >
    <BasicNav
      brandName="Author's Haven"
      logo="test-file-stub"
    />
    <div
      className="container"
      id="reset-password-container"
    >
      <h4
        id="psw-title"
      >
        RESET PASSWORD
      </h4>
      <form
        onSubmit={[Function]}
      >
        <InputField
          className="password-input"
          label="New Password"
          name="password"
          onChange={[Function]}
          type="password"
          value=""
        />
        <InputField
          className="password-input"
          label="Confirm New Password"
          name="confirmPassword"
          onChange={[Function]}
          type="password"
          value=""
        />
        <InputField
          disabled={false}
          label=""
          name=""
          style={null}
          type="submit"
          value="SAVE"
        />
        <i
          className=""
          id="spinner"
        />
      </form>
    </div>
  </div>
  <Brand />
</Fragment>
`;
